onnx/onnx.in.proto

```diff
@@ -99,8 +116,58 @@ message AttributeProto {
     STRINGS = 8;
     TENSORS = 9;
     GRAPHS = 10;
+// #if ONNX-ML
+    TUPLE = 11;
+    RECORD = 12;
+    NULLABLE = 13;
+    SPARSE = 14;
+    TYPE = 15;
+    TUPLES = 16;
+    RECORDS = 17;
+    NULLABLES = 18;
+    SPARSES = 19;
+    TYPES = 20;
+// #endif
   }
 
+// #if ONNX-ML
+  message TupleAttribute {
+    repeated AttributeProto values = 1;
+  };
+
+  message RecordAttribute {
+    repeated AttributeProto fields = 1;
```

Comment:
why not a repeated pairs (of field-name and value)?

Reply:
Because Attribute is the value representation, and it already has the name in it. That allows it to be more efficient. We'll have to add the domain and name, now that records are confirmed as nominal.

Reply:
Looking at the code, we don't need the fields property here. The values property is sufficient.